docs/docs/guides/smart_contracts/writing_contracts/notes/custom_note.md

```diff
@@ -0,0 +1,43 @@
+---
+title: Using custom notes in Aztec.nr
+---
+
+It may be useful to write a custom note if you want to use a specific type of private data or struct that does not have a default implementation in Aztec.nr. Using a custom note means that the entire note is nullified with one nullifier, which may be better for your circumstances than having multiple notes that need nullified. 
+
+If you want to work with values or addresses, you can check out [ValueNote](./value_note.md) or [AddressNote](./address_note.md). 
+
+## Define a custom note
+
+You will likely want to define your note in a new file and import it into your contract.
+
+A custom note can be defined with the macro `#[aztec(note)]` used on a struct:
+
+#include_code state_vars-CardNote noir-projects/noir-contracts/contracts/docs_example_contract/src/types/card_note.nr rust
+
+In this example, we are implementing a card note that holds a number of points as `u8`.
+
+## Implement NoteInterface
+
+You will need to implement a note interface for your note. Most of this is automatically generated by the `#[aztec(note)]` macro but can be overwritten. For your reference, this is what a note interface looks like:
+
+#include_code note_interface noir-projects/aztec-nr/aztec/src/note/note_interface.nr rust
+
+You will need to implement the functions `compute_note_hash_and_nullifier()` and `compute_note_hash_and_nullifier_without_context()` which tells Aztec how to compute nullifiers for your note.
+
+#include_code note_interface noir-projects/noir-contracts/contracts/docs_example_contract/src/types/card_note.nr rust
+ 
+In this example, these functions compute the note hash by using `compute_note_hash_for_consumption` and then generate a nullifier by hashing this note hash together with a secret.
+
+### Methods
+
+You will likely want to implement methods in order to use your note easily within contracts. For example, this may be what a `new` method can look like:
```

Comment:
```suggestion
You will likely want to implement methods in order to use your note easily within contracts. For example, this may be what a `new` method can look like, for creating a new note:
```